### PR TITLE
refactor: replace [rtl] and [ltr] selectors with :dir

### LINKED ISSUE #3050: CSS: Replace [rtl] and [ltr] selectors with :dir
- Replace our CSS [rlt] and [ltr] selector rules with :dir: https://developer.mozilla.org/en-US/docs/Web/CSS/:dir
- Deprecate getDocumentWritingMode and remove usage
- Also replace programmatic usage with css lookup (`const rtl = this.matches(':dir(rtl)')`)

### PR DESCRIPTION
**Pull request by dauriamarco**
Closes #3050

### CODE REVIEW

File: src/elements/core/a11y/arrow-navigation.spec.ts
Comment (TomMenga) [inline src/elements/core/a11y/arrow-navigation.spec.ts:1]: If the feedback  on `function getPrevAndNextKeys` is accepted, this file should be fine as it was before

**Review by TomMenga — CHANGES_REQUESTED** [review]
Good job, some minor fixes

**Review by TomMenga — APPROVED** [review]
LGTM 👍

**Review by jeripeierSBB — APPROVED** [review]
LGTM, thank you

**Pull request merged**